### DISCUSSION TITLE
how to upload file with None filename?

### DISCUSSION THREAD

**Discussion by eya46** [fastapi/fastapi#10116] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, UploadFile
from fastapi.logger import logger

app = FastAPI()


@app.post("/uploadfile")
async def create_upload_file(file: UploadFile):
    name = file.filename
    data = file.file.read()
    logger.debug(f"filename: {name}, filedata:{data}")
    return {"filename": name, "filedata": data}


if __name__ == '__main__':
    import uvicorn

    uvicorn.run("server:app")
```

### Description

> ~I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic). *~
> i don't know

how to upload file with None filename?

I checked the source code of UploadFile and found that filename can be None, 
but in fact fastapi/pydantic will report an error when receive a file without filename

https://github.com/eya46/httpx-file-test
```python
# clinet.py
import httpx, aiohttp


async def test_httpx(api: str):
    async with httpx.AsyncClient() as client:
        resp = await client.post(
            api,
            files={
                "file": (None, b"test", "text/plain"),
            }
        )
        print("-" * 20)
        print("httpx", resp)
        print("httpx", resp.json())
        print("-" * 20)


async def test_httpx_default(api: str):
    """httpx set default filename to 'upload'"""
    async with httpx.AsyncClient() as client:
        resp = await client.post(
            api,
            files={
                "file": b"test",
            }
        )
        print("-" * 20)
        print("httpx_default", resp)
        print("httpx_default", resp.json())
        print("-" * 20)


async def test_aiohttp(api: str):
    async with aiohttp.ClientSession() as client:
        resp = await client.post(
            api,
            data={
                "file": (None, b"test", "text/plain"),
            }
        )
        print("-" * 20)
        print("aiohttp", resp)
        print("aiohttp", await resp.json())
        print("-" * 20)


async def test_aiohttp_default(api: str):
    """aiohttp set default filename to 'file' ({key: file}, default filename = key)"""
    async with aiohttp.ClientSession() as client:
        resp = await client.post(
            api,
            data={
                "file": b"test",
            }
        )
        print("-" * 20)
        print("aiohttp_default", resp)
        print("aiohttp_default", await resp.json())
        print("-" * 20)


if __name__ == '__main__':
    from asyncio import run

    _api = "http://127.0.0.1:8000/uploadfile"
    run(test_httpx(_api))
    run(test_httpx_default(_api))
    run(test_aiohttp(_api))
    run(test_aiohttp_default(_api))

```

### Operating System

Windows

### FastAPI Version

0.101.1

### Pydantic Version

2.2.1

### Python Version

Python 3.11.4

### Additional Context

![image](https://github.com/tiangolo/fastapi/assets/61458340/c397d0bb-c427-4c98-91ae-79d55747d4aa)

run clinet.py
![image](https://github.com/tiangolo/fastapi/assets/61458340/122e5565-2cb4-45bf-8c9f-836bf704332a)

**Comment by romabozhanovgithub** [comment]
Hey, To upload a file with None filename, you can pass an empty string as the filename parameter when creating the UploadFile object. Here's an example:
```python
import io
from fastapi import FastAPI, UploadFile
from fastapi.logger import logger

app = FastAPI()

@app.post("/uploadfile")
async def create_upload_file(file: UploadFile):
    name = file.filename or "unknown"
    data = file.file.read()
    logger.debug(f"filename: {name}, filedata:{data}")
    return {"filename": name, "filedata": data}

if __name__ == '__main__':
    import uvicorn

    uvicorn.run("server:app")
```

**Reply by eya46** [reply]
Still throws 422😭😢
![image](https://github.com/tiangolo/fastapi/assets/61458340/a8c5d92b-e8c2-44cb-9d5f-a0fde54c9b0d)

**Comment by lraubuch** [comment, marked as answer]
This is still an open issue by the way.
My workaround was to just give the upload-file "None" as a name when sending it with httpx.
On the other hand, using requests-library there is no such problem aus you can just pass the bytes into the files parameter when using its post method.

**Reply by YuriiMotov** [reply]
Seems that there is no way to pass `none`, so the only way is to override the filename by any string.

**Reply by aspizu** [reply]
I need this because i want the user to be able to upload files using a curl command like this:
```
curl -F'file=@image.png' http://localhost:8000/
```
